llvm/include/llvm/CodeGen/TargetLowering.h

```diff
@@ -3658,6 +3696,15 @@ class TargetLoweringBase {
   /// up the MVT::VALUETYPE_SIZE value to the next multiple of 8.
   uint32_t CondCodeActions[ISD::SETCC_INVALID][(MVT::VALUETYPE_SIZE + 7) / 8];
 
+  /// For each result type and input type for the ISD::PARTIAL_REDUCE_U/SMLA
+  /// nodes, keep a LegalizeAction which indicates how instruction selection
+  /// should deal with this operation.
+  /// The key is made up of the accumulator type (AccTy) and the input type
+  /// (InTy) in the format of `(AccTy << 16) + InTy`.
+  /// If no entry exists for a given key, Expand is assumed as this
+  /// is the most common action.
+  DenseMap<uint32_t, LegalizeAction> PartialReduceMLAActions;
```

Comment:
nit: I think the key could just be `using RartialReductionTypes = std::pair<MVT::SimpleValueType, MVT::SimpleValueType>` (which is still 4-bytes), so making the key would just be `std::make_pair(AccSVT, InputSVT)`.